cpu/fe310/periph/uart.c

```diff
@@ -60,6 +62,16 @@ void uart_isr(int num)
     }
 }
 
+static void _drain(uart_t dev)
```

Comment:
I think it's possible to share this function with `_uart_isr` by adding an extra parameter for the callback. This parameter would be NULL to drain the UART and equal to `isr_ctx[dev].rx_cb` when called from `_uart_isr`. Maybe this will save a little space on ROM.
What do you think ?

Reply:
I tested locally. Even if there's less duplicated code, the resulting firmware size is increased by 18B.

Reply:
> What do you think ?

Hm, possible. Folo?